src/components/Swap/Toolbar/index.tsx

```diff
@@ -72,9 +83,23 @@ export default memo(function Toolbar() {
       }
       if (trade?.inputAmount && trade.outputAmount) {
         return impact?.warning ? (
-          <Caption.PriceImpact impact={impact} />
+          <Caption.PriceImpact
+            impact={impact}
+            expanded={expanded}
+            toggleExpanded={() => {
```

Comment:
If you alias this to a callback you can make it a one-liner:

    const onExpand = useCallback(() => {
      setExpanded((expanded) => !expanded)
    }, [])
    ...
    onToggleExpand={onToggleExpand}

Reply:
would recommend not using `useCallback` here as it adds unnecessary memory overhead whereas creating the handler function on every render is rather cheap. `useMemo` and `useCallback` should really be reserved for either expensive calculations or keeping props referentially stable

[here's](https://www.developerway.com/posts/how-to-use-memo-use-callback) a decent blog post that talks more about it